virttest/utils_time.py

```diff
@@ -0,0 +1,243 @@
+import logging
+import re
+
+from avocado.utils import process
+
+
+def get_host_timezone():
+    """
+    Get host timezone.
+
+    :return: timez_city: the host's timezone in city format, used in Linux os.
+    :return: timez_code: the hosts timezone in utc code format, used in both
+                         Linux and Win.
+    """
+    check_host_tz_cmd = 'timedatectl | grep "Time zone"'
+    timez_city = ""
+    timez_code = ""
+    try:
+        o = process.system_output(check_host_tz_cmd, timeout=240, shell=True)
+    except process.CmdError:
+        logging.warn("Can not get host's timezone")
+        return (timez_city, timez_code)
+    tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+,\s)([+|-]'
+                  r'\d{4})\)$', o, re.M | re.I)
+    if tz:
+        timez_city = tz.groups()[0]
+        timez_code = tz.groups()[1]
+    return (timez_city, timez_code)
+
+
+def sync_time_zone(os_type, vm, session, zonecode, zonecity):
+    """
+    Sync guest's timezone to host's.
+
+    This function gets guest's timezone code and compares it with host's
+    timezone code, if they are same, return True, or updates the guest's
+    timezone code to host's timezone code, then return True. If something
+    in this process, return False.
+
+    :param os_type: os type of given guest (eg. linux or windows)
+    :param vm: Virtual machine for vm.
+    :param session: VM session
+    :param zonecode: Host's timezone in code mode.
+    :param zonecity: Host's timezone in city mode.
+
+    :return: True if guest has same timezone with host.
+    """
+    status = -1
+    if os_type == "linux":
+        check_tz_cmd = 'timedatectl | grep "Time zone"'
+        set_tz_cmd = "timedatectl set-timezone %s"
+        list_tz_cmd = "timedatectl list-timezones"
+    else:
+        check_tz_cmd = "tzutil /g"
+        set_tz_cmd = 'tzutil /s "%s"'
+        list_tz_cmd = "tzutil /l"
+    o_guest = session.cmd_output(check_tz_cmd)
+    logging.info("check timezone in guest: %s" % o_guest)
+    if os_type == "linux":
+        guest_tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+'
+                            r',\s)([+|-]\d{4})\)$', o_guest, re.M | re.I)
+        if guest_tz.groups()[1] == zonecode:
```

Comment:
some code redundancy can be avoided using 
```python
timezone = get_timezone(session)
if timezone['code_timezone'] == zonecode:
    return ...
else:
    status = timezone[city_timezone]
```

Reply:
Thanks! The patterns for timezone is different in linux and windows, `get_timezone()` only accomplish one pattern for linux, so I separate it here.